src/integrationTest/sam.test.ts

```diff
@@ -397,6 +419,58 @@ describe('SAM Integration Tests', async function () {
         console.log(`DebugSessions seen in this run:\n${sessionReport}`)
     })
 
+    describe('sync', function () {
```

Comment:
TODO: move this to `sync.test.ts` and skip it when `process.env['AWS_TOOLKIT_AUTOMATION'] !== 'integration'`

Having a bunch of separate files for integration tests is unnecessary IMO.

Reply:
Also the clean-up logic I added could be improved. 404s should be treated as a 'Success' and other errors should throw. Resources that failed to delete should be persisted and retried later.

Reply:
This needs infra changes.